File: Python Implementations/main_queue.py
from linear_structures.queue import Queue
from linear_structures.sll import Node

# ==== The Main Function ====
def main() -> None:
    first_queue: Queue = Queue()
    print(first_queue)

    print("Enqueing 5 new nodes into the queue.")
    for i in range(1, 6):
        first_queue.enqueue(i * 10)
    print(first_queue)

    print("Removing the first two nodes.")
    for i in range(1, 3):
        deque_element: Node = first_queue.dequeue()
        print("Dequeue Node: ", deque_element)
        print(first_queue)

    print("Enqueing 5 more new nodes into the queue.")
    for i in range(1, 6):
        first_queue.enqueue(i * 20)
    print(first_queue)

    print("Removing the two more nodes.")
    for i in range(1, 3):
        deque_element: Node = first_queue.dequeue()
        print("Dequeue Node: ", deque_element)
        print("Peek Node: ", first_queue.peek())
    print(first_queue)

    new_queue: Queue = Queue()
    print("Empty Status of Queue 1: ", first_queue.is_empty())
    print("Empty Status of Queue 2: ", new_queue.is_empty())

    print("Clearing the Queue.")
    first_queue.delete()
    print(first_queue)

# ==== Driver Code ====
if __name__ == "__main__":
    main()

File: C++ Implementations/main_queue.cpp
#include "linear_structures/queue.h"
using namespace std;

int main(int argc, char const *argv[])
{
    Queue first_queue = Queue();
    cout << "Here is the default Queue" << endl;
    cout << first_queue;

    // Adding 10 new elements into the queue.
    cout << "Adding 10 new elements into the queue." << endl;
    for (int i = 1; i < 11; i++) {
        first_queue.enqueue(i);
    }
    cout << first_queue;

    // Removing 5 elements from the queue.
    cout << "Removing 5 elements from the queue" << endl;
    for (int i = 1; i < 5; i++) {
        cout << "Removed Element: " << first_queue.dequeue() << endl;
    }
    cout << first_queue;

    // Adding 10 new elements into the queue.
    cout << "Adding 10 new elements into the queue." << endl;
    for (int i = 1; i < 11; i++) {
        first_queue.enqueue(i * 10);
    }
    cout << first_queue;

    // Peek.
    cout << "Current Topmost Element: " << first_queue.peek() << endl;
    first_queue.dequeue();
    cout << "New Topmost Element: " << first_queue.peek() << endl;

    // Empty and Delete Ops.
    cout << "Checking for the queue to be empty " << first_queue.is_empty() << endl;
    cout << "Deleting the Queue" << endl;
    first_queue.delete_queue();
    cout << "Check for the queue to be empty " << first_queue.is_empty() << endl;
    
    return 0;
}
